domain/port/state/state.go

```diff
@@ -548,3 +585,25 @@ WHERE unit_endpoint.unit_uuid = $unitUUID.unit_uuid
 	}
 	return endpoints, nil
 }
+
+func (st *State) getUnitUUID(ctx context.Context, tx *sqlair.TX, unitName string) (unitUUID, error) {
+	unit := name{Name: unitName}
+	var uuid unitUUID
+	getUnitStmt, err := st.Prepare(`
+SELECT uuid AS &unitUUID.unit_uuid 
+FROM   unit 
+WHERE  name = $name.name
+`, unit, uuid)
+	if err != nil {
+		return unitUUID{}, errors.Errorf("preparing get unit uuid statement: %w", err)
+	}
+
+	err = tx.Query(ctx, getUnitStmt, unit).Get(&uuid)
+	if errors.Is(err, sqlair.ErrNoRows) {
+		return unitUUID{}, errors.Errorf("%w: %s", applicationerrors.UnitNotFound, unitName)
+	}
+	if err != nil {
+		return unitUUID{}, errors.Errorf("querying unit %q: %w", unitName, err)
+	}
+	return uuid, nil
+}
```

Comment:
I know we discussed this briefly, but this may be better to call the existing function on the application service to do this.

We can construct the application service for this purpose fairly easily actually. For example:
https://github.com/juju/juju/blob/eccc2eea64821b28122057b56958a7052efae50c/domain/application/modelmigration/export.go#L66-L74

Reply:
That way we can nicely drop the need to understand unit names from the ports domain entirely

Reply:
Oh nice, that looks good, will do that, handn't reliased you could get away with NotImplemented versions. Will change to this